variant-inc/terraform-variant-apps
# F/rds external secret

# Description

Updated postgres main.tf to utilize new version of 'terraform-postgres-database' and output a value specified for cake-runner. Safely removed config map functionality.


## Type of change

Please delete options that are not relevant.

(fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Utilized cake-runner(branch 'v2/refactor' with commit ID [fcf7a32] to create a demo EKS application. The demo EKS pod is named 'cake-external-secret-test-586c7b5bc6-rxwnm' in the DPL cluster. Now that config maps are removed, database secrets inserted as environment variables are now marked as sensitive variables.

<img width="695" alt="Screen Shot 2022-08-10 at 11 43 39 AM" src="https://user-images.githubusercontent.com/97901744/183949293-ef6414ed-2db7-4e4d-a463-aedeaee40867.png">

Testing Steps
1) Clone 'cake-runner' repo.
2) Create a '.env' file with the following contents in the base directory
```
AWS_DEFAULT_REGION=us-east-1
AWS_REGION=us-east-1
S3_BUCKET=lazy-tf-state-0k3nc997arlf7k1a
DYNAMO_DB_TABLE=lazy_tf_state
#CAKE_DESTROY=true
```
3) In '/hack/spec.yaml', I personally used the following spec.yaml. Any will work fine as long as the postgres infrastructure is being provisioned for this test.
```
name: cake-external-secret-test
octopus:
  space: DevOps
  group: sre
git:
  repository: demo-python-flask-variant-api
  user: vibin
  version: 0.1.0-samir-tests0001.885
  image: 064859874041.dkr.ecr.us-east-1.amazonaws.com/demo/python-flask-variant-api:0.1.0-samir-tests0001.496
  isinfrastructure: false
tags:
  owner: sre
  team: sre
  purpose: test
api:
  service:
    targetPort: 5000
infrastructure:
  postgres:
    - name: dso-test-rds-external
      role_name: dso-rds-user-external
      reference: dso-rds-external
```
4) Run the following command. Ensure '../../../terraform-variant-apps' is referencing to a clone of the repository 'terraform-variant-apps' and '../../hack/spec.yaml' is referencing the spec.yaml.
```dotnet run --project ./src/Variant.Cake.Runner/ -- --deployYamlFilePath ../../hack/spec.yaml --target Terraform --logLevel Debug --tfModulesDirPath ../../../terraform-variant-apps```

5) Check whether a pod was provisioned in the cluster.


## Checklist
